fvm/blueprints/system.go

```diff
@@ -21,9 +22,17 @@ transaction {
 `
 
 // SystemChunkTransaction creates and returns the transaction corresponding to the system chunk
-// at the specified service address.
-func SystemChunkTransaction(serviceAddress flow.Address) *flow.TransactionBody {
-	return flow.NewTransactionBody().
-		SetScript([]byte(fmt.Sprintf(systemChunkTransactionTemplate, serviceAddress))).
-		AddAuthorizer(serviceAddress)
+// for the given chain.
+func SystemChunkTransaction(chain flow.Chain) (*flow.TransactionBody, error) {
+
+	contracts, err := systemcontracts.SystemContractsForChain(chain.ChainID())
```

Comment:
Was `systemcontracts.SystemContractsForChain` already there but not used? Its not part of this PR

Reply:
It was there, just wasn't added to the system chunk transaction